koku/api/iam/view/user_preference.py

```diff
@@ -0,0 +1,272 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""View for User Preferences."""
+
+from rest_framework import exceptions, mixins, viewsets
+from rest_framework.authentication import (SessionAuthentication,
+                                           TokenAuthentication)
+from rest_framework.permissions import IsAuthenticated
+
+import api.iam.models as models
+import api.iam.serializers as serializers
+from api.common.permissions.owner import IsOwner
+
+
+class UserPreferenceViewSet(mixins.CreateModelMixin,
+                            mixins.DestroyModelMixin,
+                            mixins.ListModelMixin,
+                            mixins.RetrieveModelMixin,
+                            mixins.UpdateModelMixin,
+                            viewsets.GenericViewSet):
+    """User Preference View.
+
+    A viewset that provides default `create()`, `destroy`, `retrieve()`,
+    and `list()` actions.
+    """
+
+    lookup_field = 'uuid'
+    queryset = models.UserPreference.objects.all()
+    serializer_class = serializers.UserPreferenceSerializer
+    authentication_classes = (TokenAuthentication,
+                              SessionAuthentication)
+    permission_classes = (IsOwner, IsAuthenticated)
+
+    def get_queryset(self):
+        """Get a queryset that only displays the owner's preferences."""
+        return self.queryset.filter(user=self.request.user)
+
+    def perform_create(self, serializer):
+        """Create a preference."""
+        serializer.save()
+
+    def _validate_user(self, user_id, uuid):
+        """Validate that the user object is assigned the given uuid."""
+        user_a = models.User.objects.get(id=user_id)
+        user_b = models.User.objects.get(uuid=uuid)
+        return user_a == user_b
+
+    def create(self, request, *args, **kwargs):
+        """Create a user preference.
+
+        @api {post} /api/v1/user/:id/preferences/ Create a user preference
+        @apiName createUserPreference
+        @apiGroup UserPreferences
+        @apiVersion 1.0.0
+        @apiDescription Create a user preference.
+
+        @apiHeader {String} token User authorizaton token.
+        @apiHeaderExample {json} Header-Example:
+            {
+                "Authorizaton": "Token 45138a913da44ab89532bab0352ef84b"
+            }
+
+        @apiParam {Number} id User unique ID.
+
+        @apiParam (Request Body) {String} preference a stringified JSON object
+        @apiParamExample {json} Request Body:
+            {
+                "name": "my-preference-name",
+                'preference': {'test-pref': ['a', ['b', 'c'], {'foo': 'bar'}]},
```

Comment:
Let's use an example for a preference we want as I don't think we want store any random JSON for preferences.